docs/DeveloperGuide.md

```diff
@@ -573,47 +573,275 @@ testers are expected to do more *exploratory* testing.
+
+    1. Test Case: `edit 1`<br>
+       Expected: First contact has no fields updated. Error details are shown in the status message. Must specify fields to edit.
+
+    1. Test Case: `edit 0`<br>
+       Expected: No fields are updated at all across all contacts. Error details are shown in the status message. Index must be positive integer.
+
+1. Editing a person while no persons are being shown.
+
+    1. Prerequisites: List of contacts contains no persons.
+
+    1. Test Case: `edit 1 p/98765432`<br>
+       Expected: No fields are updated. Error details are shown in the status message. There must be persons in the list of contacts to edit.
 
-1. _{ more test cases …​ }_
 
 ### Deleting a person
 
-1. Deleting a person while all persons are being shown
+1. Deleting a person while persons are being shown successfully
+
+   1. Prerequisites: At least one person in the list of contacts. Either from running `list` command or previous `find` command.
+
+   1. Test Case: `delete 1`<br>
+      Expected: First contact is deleted from the list. Details of the deleted contact shown in the status message. Deleting by index.
+
+   1. Test Case: `delete n/[VALID NAME]`<br>
+      Expected: The contact with the matching `NAME` is deleted from the list. Details of the deleted contact shown in the status message. Deleting by valid name.
+
+1. Deleting a person while persons are being shown unsuccessfully.
+
+   1. Prerequisites: At least one person in the list of contacts. Either from running `list` command or previous `find` command.
+
+   1. Test Case: `delete 0`<br>
+      Expected: No person is deleted. Error details shown in the status message. Deleting by wrong index.
+
+   1. Test Case: `delete n/[INVALID NAME]`<br>
+      Expected: No person is deleted. Error details shown in the status message. Deleting by wrong name.
+
+   1. Other incorrect delete commands to try: `delete`, `delete x`, `...` (where x is larger than the list size)<br>
+      Expected: Similar to previous.
+
+1. Deleting a person while no persons are being shown
+
+   1. Prerequisites: List of contacts contains no persons.
+
+   1. Test Case: `delete n/[VALID NAME]`<br>
+      Expected: No contacts are deleted. Error details are shown in the status message. Deleting when no contacts are shown.
+
+   1. Test Case: `delete 1`<br>
+      Expected: No contacts are deleted. Error details are shown in the status message. Deleting when no contacts are shown.
+
+
+### Updating insurance
+
+1. Updating a person's insurance while persons are being shown successfully
+
+   1. Prerequisites: At least one person in the list of contacts. Either from running `list` command or previous `find` command.
+
+   1. Test Case: `insurance 1 hi/`<br>
+      Expected: First contact is shown to have health insurance. Details of the updated insurance policy are shown in the status message. Updating single insurance.
```

Comment:
Perhaps we could add on that the first contact is also shown to not have any of the other 3 insurances?

Reply:
Fixed thank you 👍